lib/routes/routes-qr.js

```diff
@@ -0,0 +1,108 @@
+'use strect'
+
+const ecc = require('eosjs-ecc')
+
+const {
+  responder,
+  checkPayloadByJsonSchema,
+  isMtsOlderThan
+} = require('./helpers')
+const { SignVerificationError } = require('./errors')
+
+class RoutesQR {
+  constructor (conf, plugins, opts) {
+    this.conf = conf
+    this.plugins = plugins
+    this.opts = opts
+
+    this.expSec = 20 * 60
+    this.redis = this.plugins.simpleWalletProtocolRedisDb
+  }
+
+  verifySign ({
+    timestamp,
+    sign,
+    uuID,
+    account,
+    ref
+  }) {
+    if (
+      timestamp &&
+      sign &&
+      uuID &&
+      account &&
+      ref &&
+      typeof sign === 'string' &&
+      typeof uuID === 'string' &&
+      typeof account === 'string' &&
+      typeof ref === 'string'
+    ) {
+      const data = timestamp + account + uuID + ref
+      const pubkey = ecc.recover(sign, data)
+      const isValid = ecc.verify(sign, data, pubkey)
```

Comment:
this will fail with different chain ids

Reply:
as is mentioned previously

best to take a look how we handle signed trasnactions with TOS in moonbeam

might be a good example how we verify presigned txs

Reply:
maybe i am missing something here.. i.e. how do we verify that they are `@active` or `@owner`?

Reply:
wouldn't our standard login be exactly like https://github.com/southex/SimpleWallet/blob/master/README_en.md#scenario-2-the-mobile-app-of-dapp-pulls-up-the-wallet-app-and-requests-the-login-authorization

(scenario 2, without the polling) 


this could be just reused then?

and how is this goign to sign other transactions later, apart from login?

Reply:
please also include an example how you would sign:

1.  a eos transfer with this method (i.e. a trade on eosfinex)
2. a call to `g-tos`